aws/lambda_function.py

```diff
@@ -0,0 +1,174 @@
+    sys.exit()
+
+logger.info("SUCCESS: Connection to RDS Postgres instance succeeded")
+
+def wrangle(df):
+    print(f"Original shape: {df.shape}")
+    clean_df = df.copy()
+    # NOT CURRENT IN GUESTS DB
+    # columns = ["private_disability_income",'predicted_exit_destination']
+    clean_df.columns = map(str.lower, clean_df.columns)
+    clean_df.drop(columns="index", inplace=True)
+    # Drop sensitive columns
+    columns = ['enrollment_created_by',	'first_name',	'last_name',	
+                'ssn', 'dob']
+    clean_df.drop(columns=columns, inplace=True)
+    # drop reviewed columns
+    drop_col = ["days_enrolled_in_project","bednights_during_report_period", 
+                "entire_episode_bednights", "contact_services", 
+                "non-cash_count_at_exit", "info_release_status",
+                "other_public", "unemployement_income", "housing_checkins"]
+    clean_df.drop(columns=drop_col, inplace=True)
+    # Drop questionable columns
+    # REMINDER: important, missing values, these should be required
+    columns = ["zip", "income_at_entry", "income_at_exit", 
+            "length_of_time_homeless", "homeless_start_date",
+            "vet_status", "living_situation", "length_of_stay",
+            "homeless_start_date", "times_homeless_last_3years",
+            "total_months_homeless", "last_perm_address", "state",
+            "municipality", "housing_status", "domestic_violence",
+            "currently_fleeing", "when_dv_occured", "engagement_date",
+            "last_grade_completed", "school_status", "employed_status",
+            "reason_not_employed", "type_of_employment", "looking_for_work",
+            "soar_eligibility", "alcohol_abuse", "chronic_health_condition",
+            "developmental_disability", "substance_abuse",
+            "mental_health_problem", "physical_disability"]
+    clean_df.drop(columns=columns, inplace=True)
+    # Drop interesting most_recent columns, check in with @J, future feature engineering
+    columns = ["most_recent_rrh", "most_recent_street_outreach",
+                "most_recent_ce", "most_recent_es", "most_recent_trans",
+                "most_recent_psh", "most_recent_prevention"]
+    clean_df.drop(columns=columns, inplace=True)
+    # Drop interesting datetime columns, check in with @J, future feature engineering
+    columns = ["date_of_last_contact", "date_of_first_contact", 
+                "date_of_last_stay", "date_of_first_stay"]
+    clean_df.drop(columns=columns, inplace=True)
+    # Drop unecessary columns
+    # lat/lon (not useful), income_at_update (extreme low count), program_type (constant)
+    # client_id (constant), chronic_homeless_status_assessment (identical exists),
+    # org_name (constant), project_type (constant), fed_grant_programs (constant),
+    # client_location (constant), days_enrolled_until_rrh_movein (constant),
+    # days_enrolled_until_engaged (difficult), current_status (?), connected_to_mvento (?),
```

Comment:
I believe it will be worth the effort to incorporate 'days enrolled until engaged'.  This is an engineered column that calculates how many days passed between their date of enrollment and their first case management.  We might see a trend in predictions that make people with case management meetings within the first week of being a guest with us versus guests that managed to stay 3 plus weeks before their first case management.  

Connected to mvento indicates whether or not a school aged child is connected to their school's McKinny-Vento liason.  McKinny Vento is a law that passed that gives school aged children certain rights.  One example is that children can stay in their school they were enrolled in prior to becoming homeless.  If they are connected to mvento, they can get free transportation to and from their school from the shelter.

Reply:
This iteration of the model is for at-entry only, we haven't taken into account modifications for accurate prediction once is guest has been in the system for awhile.